src/main/java/com/google/sps/servlets/FetchQuestionsServlet.java

```diff
@@ -35,19 +35,29 @@
 import javax.servlet.http.HttpServletResponse;
 
 /** 
- * This servlet will retrieve forum posts to be displayed on the page.
+ * This servlet will retrieve questions to be displayed on the page.
  */
-@WebServlet("/fetch-forum")
-public class FetchForumServlet extends HttpServlet {
+@WebServlet("/fetch-questions")
+public class FetchQuestionsServlet extends HttpServlet {
 
   /** 
-   * This method will get the forum questions from the query and return them as a JSON string.
+   * This method will get the questions from the query and return them as a JSON string.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     List<QuestionObject> questions = new ArrayList<>();
     
-    String query = Utility.fetchQuestionQuery;
+    String query;
+
+    // ID of the question to query. -1 means that all questions are to be queried.
+    int question_id = Integer.parseInt(request.getParameter("id"));
+
+    if (question_id != -1) {
+      // Condition to fetch only one question.
```

Comment:
Consider adding a constant in the SQL Constants to represent the -1 to further explain the meaning of this number.
```suggestion

    if (question_id != SqlConstants.USER_NOT_LOGGED_IN) {
      // Condition to fetch only one question.
```

Reply:
done